docs/contributing/debugging.md

```diff
@@ -0,0 +1,33 @@
+### Debugging
+
+Debugging java agent can be a challenging task since some instrumentation
+code is directly inlined into target classes and debugger is
+usually not attached early enough to activate breakpoints
+in agent initialization code (`premain`/`agentmain`, `agent-tooling` artifact etc.).
```

Comment:
`agentmain` is for "hot attach" and I think breakpoints may work there (though we don't use/test/doc hot attach at this time).

naming specific classes as examples I think is good, since lots of `agent-tooling` classes will work under debugger, just not the initialization code

```suggestion
in agent initialization code (`OpenTelemetryAgent`, `AgentInitializer`, `AgentInstaller`, `TracerInstaller`, etc.).
```